### PR TITLE
Save product details to database

### PR DESCRIPTION
**Pull request by kamrulbari14**
-- ApiController
-- BeanConfig
-- ProductController
-- BaseDto
-- ProductDto
-- Make created at not updatable in BaseEntity
-- Rename sellablePrice to saleablePrice in Product
-- ProductRepository
-- ProductService and ProductServiceImpl

### CODE REVIEW

File: src/main/java/com/inventory/demo/dto/ProductDto.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+@Setter
+@Getter
+@NoArgsConstructor
+@AllArgsConstructor
+public class ProductDto extends BaseDto{
+    private String productName;
+    private String batchId;
+    private Double originalPrice;
+    private Double salablePrice;
+    private Double totalQuantity;
```
Comment (akhter-rakib) [inline src/main/java/com/inventory/demo/dto/ProductDto.java]: Need to add the Bean validation for user Input

File: src/main/java/com/inventory/demo/entity/Product.java [hunk 2]
```diff
@@ -10,8 +10,8 @@
 @Entity
 public class Product extends BaseEntity {
     private String productName;
+    private String batchId;
     private Double originalPrice;
-    private Double sellablePrice;
+    private Double salablePrice;
     private Double totalQuantity;
```
Comment (akhter-rakib) [inline src/main/java/com/inventory/demo/entity/Product.java]: Need to add the JPA Bean validation for user Input

File: src/main/java/com/inventory/demo/service/impl/ProductServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    private final ModelMapper modelMapper;
+
+    public ProductServiceImpl(ProductRepository productRepository, ModelMapper modelMapper) {
+        this.productRepository = productRepository;
+        this.modelMapper = modelMapper;
+    }
+
+    @Override
+    public Response saveProduct(ProductDto productDto) {
+        Product product = productRepository.findByBatchIdAndIsActive(productDto.getBatchId(), Boolean.TRUE);
```
Comment (akhter-rakib) [inline src/main/java/com/inventory/demo/service/impl/ProductServiceImpl.java]: Can ignore the validation with batch. Apply product name validation only

**Review by akhter-rakib — CHANGES_REQUESTED** [review]
Please have a look at the comment and change accordingly.

File: src/main/java/com/inventory/demo/dto/ProductDto.java (hunk 1 above)
Comment (kamrulbari14) [reply src/main/java/com/inventory/demo/dto/ProductDto.java]: Required Validation added

File: src/main/java/com/inventory/demo/entity/Product.java (hunk 2 above)
Comment (kamrulbari14) [reply src/main/java/com/inventory/demo/entity/Product.java]: Required Validation added

File: src/main/java/com/inventory/demo/service/impl/ProductServiceImpl.java (hunk 3 above)
Comment (kamrulbari14) [reply src/main/java/com/inventory/demo/service/impl/ProductServiceImpl.java]: Removed the batch Id property and the validation

**Pull request merged**